pybamm/models/submodels/particle_mechanics/base_mechanics.py

```diff
@@ -43,12 +43,19 @@ def _get_mechanical_results(self, variables):
         sto_rav = variables[f"R-averaged {domain} particle concentration"]
         c_s_surf = variables[f"{Domain} particle surface concentration [mol.m-3]"]
         T_xav = variables["X-averaged cell temperature [K]"]
+        phase_name = self.phase_name
+        T = pybamm.PrimaryBroadcast(
+            variables[f"{Domain} electrode temperature [K]"],
+            [f"{domain} {phase_name}particle"],
+        )
         eps_s = variables[f"{Domain} electrode active material volume fraction"]
 
-        Omega = domain_param.Omega
+        #use a tangential approximation for omega
+        sto = variables[f"{Domain} particle concentration"]
+        Omega = pybamm.r_average(domain_param.Omega(sto))
```

Comment:
while we're here, could Omega also depend on T?

Reply:
No, that would be a thermal expansion coefficient, generally called `α`

Reply:
I mean I guess they're the same thing really, but in some ways that's an entirely different model.

Reply:
I'll do it anyway